### ISSUE TITLE
Generating long sequences - GAN mode collapse  

### ISSUE THREAD

**Issue by Rane90** [ydataai/ydata-synthetic#181]
Hi,

Thank you for uploading this code!

I've been trying to use your tutorial for time series generation to generate raw EEG signals. In this domain the sequence legnth has to be much larger than the proposed 24, otherwise, the data is not informative.

 I've been working with data of shape (N, 2, 1500) and the output seems to be in mode collapse, i.e. all outputs look the same. After some hyper-parameter optimization, and modifying the generator loss to Wasserstein loss, I've managed to make the two channels differ from one another. Nevertheless, the outputs are almost all the same, meaning that my model is still in mode collapse.

Is there anything you recommend doing? @fabclmnt [your post](https://towardsdatascience.com/synthetic-time-series-data-a-gan-approach-869a984f2239) indicates that it is possible to generate EEG signals. Do you have any ready implantation?

Thank you :)

**Comment by pbezz1** [comment]
What's the shape of your original dataset like?  How you tried different batch sizes?

**Comment by Rane90** [comment]
Thank you for your response.

The (2, 1500) samples are cut out of long sequences of EEG recordings. Hours, with frequcy=512hz. So the real shape of the data is much larger.

And yes, I have tried larger batches and they helped. Currently, I'm fixated on 1024.

**Comment by pbezz1** [comment]
So you have 2 input data features and 1500 data rows correct?

The batch size may improve execution time but can degrade the performance of your GAN. I suggest you run TimeGAN changing the batch size only with each run and monitor the discrimination and prediction scores. You might also want to add some loss graphs to monitor which part of the training is collapsing. 

TimeGAN is a complex architecture with a lot of hyperparameters. I've done some experiments with TimeGAN with financial time series data using price series of many stocks together, changing the batch size and the sequence length. 

Balancing data quality with performance is not an easy task, you just have to keep on experimenting with a selection of parameters. You might also want to change the optimisers from Adam to RMSProp and play around a bit with the learning rate.

Hope this helps.

**Comment by Rane90** [comment]
Thank you so much. I'll try to follow your advice.
Indeed this is not an easy task.

**Comment by fabclmnt** [comment]
Hi @Rane90,

TimeGAN architecture can be applicable to a variety of TimeSeries datasets, nevertheless the hyper parameter tuning for such an architecture will vary a lot from dataset to dataset. 

The architecture is ready to be used, but you need to tweak and play around with the parameters. Also, bear in mind the architecture was build to work for time sequences/windows. Have a look into the the supplementary materials available for TimeGan: https://www.vanderschaar-lab.com/papers/NIPS2019_TGAN_Supplementary.pdf.
To generate longer sequences,  I would advise you to use a different architecture.

Number of iterations, Learning rate an applied data preparation are usually the parameters that have an higher effect on the results you will get.

**Comment by fabclmnt** [comment]
I'll be happy to follow-up any other questions at our community slack.
